pkgs/applications/version-management/gitlab/gitlab-shell/default.nix

```diff
@@ -1,41 +1,61 @@
-{ stdenv, ruby, bundler, fetchFromGitLab, go }:
-
-stdenv.mkDerivation rec {
-  version = "9.3.0";
-  pname = "gitlab-shell";
+{ stdenv, ruby, bundler, fetchFromGitLab, buildGoPackage, bundlerEnv }:
 
+let
+  version = "10.0.0";
   src = fetchFromGitLab {
     owner = "gitlab-org";
     repo = "gitlab-shell";
     rev = "v${version}";
-    sha256 = "1r000h4sgplx7giqvqs5iy0zh3drf6qa1iiq0mxlk3h9fshs1348";
+    sha256 = "0n1llkb0jrqxm10l9wqmqxjycydqphgz0chbbf395d8pywyz826x";
   };
+  rubyEnv = bundlerEnv {
+    name = "gitlab-shell-env";
+    inherit ruby;
+    gemdir = ./.;
+  };
+  goPackage = buildGoPackage {
+    pname = "gitlab-shell-go";
+    inherit src version;
+
+    patches = [ ./remove-hardcoded-locations.patch ];
 
-  buildInputs = [ ruby bundler go ];
+    goPackagePath = "gitlab.com/gitlab-org/gitlab-shell";
+    goDeps = ./deps.nix;
 
+    postConfigure = ''
+      rm -r "$NIX_BUILD_TOP/go/src/gitlab.com/gitlab-org/gitaly/vendor/"
```

Comment:
This deserves a comment. I suspect this is what I was missing in #68908.

Reply:
It did indeed solve that issue. :) Good call, I'll add comments.